src/main/java/org/jabref/gui/exporter/SaveDatabaseAction.java

```diff
@@ -216,48 +220,45 @@ private boolean save(Path targetPath, SaveDatabaseMode mode) {
     }
 
     private boolean saveDatabase(Path file, boolean selectedOnly, Charset encoding, SavePreferences.DatabaseSaveType saveType) throws SaveException {
+
+
         SavePreferences preferences = this.preferences.getSavePreferences()
                                                       .withEncoding(encoding)
                                                       .withSaveType(saveType);
-        try (AtomicFileWriter fileWriter = new AtomicFileWriter(file, preferences.getEncoding(), preferences.shouldMakeBackup())) {
-            BibtexDatabaseWriter databaseWriter = new BibtexDatabaseWriter(fileWriter, preferences, entryTypesManager);
 
-            if (selectedOnly) {
-                databaseWriter.savePartOfDatabase(panel.getBibDatabaseContext(), panel.getSelectedEntries());
-            } else {
-                databaseWriter.saveDatabase(panel.getBibDatabaseContext());
-            }
-
-            panel.registerUndoableChanges(databaseWriter.getSaveActionsFieldChanges());
+        Consumer<List<FieldChange>> consumer = fieldChanges -> panel.registerUndoableChanges(fieldChanges);
```

Comment:
I would move the consumer to the `create` method.

Reply:
Isn't that an architecture violation then?